### ISSUE TITLE
[Bug]: triplot duplicates label legend  

### ISSUE THREAD

**Issue by maweigert** [matplotlib/matplotlib#22965]
### Bug summary

Hi, 

When using `triplot` with a `label` the resulting legend contains two duplicated entries (even though the markers is None):

   
```python 

plt.triplot([0,2,1],[0,0,1],[[0,1,2]], label='Label'); 
plt.legend()
```
<img width="500" alt="Screenshot 2022-05-03 at 12 17 10" src="https://user-images.githubusercontent.com/11042162/166437380-3c5a22bd-e404-4224-9907-288f97ff75c5.png">

I think expected behavior would be either 

1) Remove the second legend entry if no markers are given, or
2) Remove the second legend entry altogether, e.g. by using `kw_markers.pop('label',None)` around here https://github.com/matplotlib/matplotlib/blob/2aa71eb3559af56c56b9d137790107bffd83cddd/lib/matplotlib/tri/triplot.py#L78

### Code for reproduction

```python
see above
```

### Actual outcome

see above

### Expected outcome

see above

### Operating system

MacOS

### Matplotlib Version

3.6.0.dev2140+g2aa71eb355.d20220503

### Matplotlib Backend

QtAgg

### Installation

pip

**Comment by timhoffm** [comment]
In my understanding, there should always be a visible line. Otherwise, there's no point in triangulating and one could have used a simpler `plot` or `scatter`. Thus, we can simply pass a given label to the line artist and suppress it for the marker artist. It should be very rare that one wants to label the marker as well - confirmed by the fact that nobody has asked for that so far. If really needed they can still use `set_label()` on the marker artist later.

**Comment by oscargus** [comment]
Marking this as a good first issue. Solution is to either pop `label` from `kw` or set the `label` to `'_nolegend_'` (or whatever the default name is). Plus write test and possibly a release note.

https://github.com/matplotlib/matplotlib/blob/2aa71eb3559af56c56b9d137790107bffd83cddd/lib/matplotlib/tri/triplot.py#L74-L78

**Comment by amartinez1224** [comment]
I want to help with this, can I have the issue assigned to me?

**Comment by oscargus** [comment]
@amartinez1224 We do not assign issues, but feel very free and encouraged to approach this despite that. (The fact that you have shown interest will most probably act like an "assign" and encourage others to look elsewhere.)

**Comment by amartinez1224** [comment]
**Test**
By explicitly listing the artists and labels in the legend there are no duplicated entries.
```
ax.triplot([0,2,1],[0,0,1],[[0,1,2]])
ax.legend(['label'])
```
<img src="https://user-images.githubusercontent.com/44838230/188783881-a98c1b85-4b93-4ea4-8595-c8dafd2dcbcd.png" alt="triplot" width="30%"/>

Using `@check_figures_equal`, I added a test that compares the `triplot` that uses `label` with the plot described above. Is this test enough or should I test something else?

**Comment by amartinez1224** [comment]
I changed the test to the explicit logic test as suggested by @timhoffm in https://github.com/matplotlib/matplotlib/pull/23813#discussion_r964479318.

**Comment by timhoffm** [comment]
Closed by #23813.
